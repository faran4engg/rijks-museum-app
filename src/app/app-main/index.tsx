import { Route, BrowserRouter } from 'react-router-dom';
import { QueryCache, QueryClient, QueryClientProvider } from 'react-query';
import { ReactQueryDevtools } from 'react-query/devtools';
import { LayoutContainer } from 'app/domains/Layout/containers/async';
import './index.css';

const queryCache = new QueryCache();
const queryClient = new QueryClient({
  queryCache,
});

const AppMain = () => {
  return (
    <QueryClientProvider client={queryClient}>
      <BrowserRouter>
        <Route path="/" component={LayoutContainer} />
      </BrowserRouter>
      <ReactQueryDevtools initialIsOpen={false} />
    </QueryClientProvider>
  );
};

export default AppMain;
