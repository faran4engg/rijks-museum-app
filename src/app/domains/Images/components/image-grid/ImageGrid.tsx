import { FC } from 'react';
import { CardSkeleton } from 'app/domains/Common/components/loaders';
import { ImageGridProps } from './types';
import ImageThumbnail from 'app/domains/Images/components/image-thumbnail/ImageThumbnail';
import NotFoundPage from 'app/pages/NotFoundPage';

const errorCls = 'flex items-center justify-center max-w-sm mx-auto mt-8';

const ImageGrid: FC<ImageGridProps> = ({ isLoading, images = [] }) => {
  const Images = (
    <div className={`masonry-with-columns`}>
      {images?.map(({ principalOrFirstMaker, title, webImage: { url } }) => (
        <ImageThumbnail
          key={title}
          url={url}
          authorName={principalOrFirstMaker}
          title={title}
        />
      ))}
    </div>
  );

  return (
    <>
      {!isLoading && !images.length && (
        <div className={errorCls}>
          <NotFoundPage />
        </div>
      )}

      {isLoading ? <CardSkeleton noOfCards={3} /> : Images}
    </>
  );
};

export default ImageGrid;
