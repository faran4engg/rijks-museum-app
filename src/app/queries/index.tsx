import { ImagesResponseAPI } from 'app/kernel/types';
import { axiosInstance } from '../axios/axios-instance';

const { REACT_APP_API_PUBLIC_KEY } = process.env;

const getImages = async (): Promise<ImagesResponseAPI> =>
  axiosInstance.get(
    `/collection?key=${REACT_APP_API_PUBLIC_KEY}&involvedMaker=Rembrandt+van+Rijn`,
  );

export { getImages };

// Willem Drost
// Moses ter Borch
// Adam von Bartsch
// Jacob Houbraken
