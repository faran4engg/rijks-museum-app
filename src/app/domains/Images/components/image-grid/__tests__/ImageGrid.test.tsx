import { create } from 'react-test-renderer';

import ImageGrid from '../ImageGrid';
import { imagesResponseMock } from 'app/mock';

jest.mock('app/pages/NotFoundPage', () => 'NotFoundPage');
jest.mock('app/domains/Common/components/loaders', () => ({
  CardSkeleton: () => 'CardSkeleton',
}));

jest.mock(
  'app/domains/Images/components/image-thumbnail/ImageThumbnail',
  () => 'ImageThumbnail',
);

const images = imagesResponseMock.artObjects;

describe('ImageGrid', () => {
  it('should render CardSkeleton when loading', () => {
    const component = create(<ImageGrid isLoading />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should render NotFoundPage when its not loading and no image data received', () => {
    const component = create(
      <ImageGrid isLoading={false} images={[]} />,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should render images', () => {
    const component = create(
      <ImageGrid isLoading={false} images={images} />,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
