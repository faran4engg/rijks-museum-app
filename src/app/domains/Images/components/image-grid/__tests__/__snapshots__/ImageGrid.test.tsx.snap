// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageGrid should render CardSkeleton when loading 1`] = `"CardSkeleton"`;

exports[`ImageGrid should render NotFoundPage when its not loading and no image data received 1`] = `
Array [
  <div
    className="flex items-center justify-center max-w-sm mx-auto mt-8"
  >
    <NotFoundPage />
  </div>,
  <div
    className="masonry-with-columns"
  />,
]
`;

exports[`ImageGrid should render images 1`] = `
<div
  className="masonry-with-columns"
>
  <ImageThumbnail
    authorName="Rembrandt van Rijn"
    title="Isaak en Rebekka, bekend als ‘Het Joodse bruidje’"
    url="https://lh3.googleusercontent.com/mAyAjvYjIeAIlByhJx1Huctgeb58y7519XYP38oL1FXarhVlcXW7kxuwayOCFdnwtOp6B6F0HJmmws-Ceo5b_pNSSQs=s0"
  />
  <ImageThumbnail
    authorName="Rembrandt van Rijn"
    title="De Nachtwacht"
    url="https://lh3.googleusercontent.com/0cgLYKx0RS2R0RkCr9z29qGHmQMIchfN3Mqpvfx9msLqxVdAiNJxNN9le3GnFF6kUrPG84DWNFBqLXFM7OOqSZFJsVnvxrDu-PlK3ars=s0"
  />
  <ImageThumbnail
    authorName="Rembrandt van Rijn"
    title="Portret van een vrouw, mogelijk Maria Trip"
    url="https://lh3.googleusercontent.com/AyiKhdEWJ7XmtPXQbRg_kWqKn6mCV07bsuUB01hJHjVVP-ZQFmzjTWt7JIWiQFZbb9l5tKFhVOspmco4lMwqwWImfgg=s0"
  />
  <ImageThumbnail
    authorName="Rembrandt van Rijn"
    title="Zelfportret als de apostel Paulus"
    url="https://lh3.googleusercontent.com/NrCcfeY0r2F3M2hIQe5SLDRofR2tVzeOH18VjflOYGj88v4clb4v2H_VgCZR4nJhYsxxH9ATzfkL2tRqOWEK5-gPVEE=s0"
  />
  <ImageThumbnail
    authorName="Rembrandt van Rijn"
    title="De waardijns van het Amsterdamse lakenbereidersgilde, bekend als ‘De Staalmeesters’"
    url="https://lh3.googleusercontent.com/gShVRyvLLbwVB8jeIPghCXgr96wxTHaM4zqfmxIWRsUpMhMn38PwuUU13o1mXQzLMt5HFqX761u8Tgo4L_JG1XLATvw=s0"
  />
  <ImageThumbnail
    authorName="Rembrandt van Rijn"
    title="Zelfportret"
    url="https://lh3.googleusercontent.com/7qzT0pbclLB7y3fdS1GxzMnV7m3gD3gWnhlquhFaJSn6gNOvMmTUAX3wVlTzhMXIs8kM9IH8AsjHNVTs8em3XQI6uMY=s0"
  />
  <ImageThumbnail
    authorName="Rembrandt van Rijn"
    title="Oopjen Coppit"
    url="https://lh3.googleusercontent.com/ib8SNTK2Qk-z64UYuu-_mI3FswMpYmmNU871wu5diDEPyjxmYJcNI4qRtqxlvKkVnrXTAxAFkuHX7DAN9ZwPFzS5fGE=s0"
  />
  <ImageThumbnail
    authorName="Rembrandt van Rijn"
    title="Marten Soolmans"
    url="https://lh3.googleusercontent.com/hDN8_JS4BbbksBXkxIDUldJvooejHBfQjgD3b_Jr73HjVOAs8thpnvCJA3WUtQFxlbjx-QB1-IyqhgEiM4dFLFk1Sg=s0"
  />
  <ImageThumbnail
    authorName="Rembrandt van Rijn"
    title="Salome ontvangt het hoofd van Johannes de Doper"
    url="https://lh3.googleusercontent.com/sB8bcd5NQmryNNIZ3-xb27VMtsohrhjMxm3HZMX-ajffrWUyKh7WoBzOik5k3rc-8XNzmDrpNfyErwrsbN2bs2o63EQ=s0"
  />
  <ImageThumbnail
    authorName="Rembrandt van Rijn"
    title="Portret van Haesje Jacobsdr van Cleyburg"
    url="https://lh3.googleusercontent.com/8IJF23RpZ5Jjukz6GHkAvagBGK4X5ZRJygm6DMvR7a0eQQWc_NoPzDZmf2UugJ-2NUxgFcJ2tx0c26K8EZvzkQBlra3o=s0"
  />
</div>
`;
